rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/BatchJobInfrastructure.java

```diff
@@ -489,12 +490,12 @@ public void configure(Object... parameters) {
                 this.nodeTimeOut = 1000 * 60 * 5;
             }
             this.serverName = parameters[index++].toString();
-            if (parameters[index] == null) {
-                throw new IllegalArgumentException("Credentials must be specified");
-            }
             try {
-                persistedInfraVariables.put(CREDENTIALS_KEY,
-                                            Credentials.getCredentialsBase64((byte[]) parameters[index++]));
+                Object possibleCredentials = parameters[index++];
```

Comment:
Shouldn't you verify at that point that the provided objected is castable in Credential ? It could be a fast-fail way to ensure the user-provided parameter is valid.